src/contracts/fund/hub/Hub.ts

```diff
@@ -164,7 +193,25 @@ export class Hub extends Contract {
    * @param from The sender address.
    */
   public setRouting(from: Address) {
-    return this.createTransaction('setRouting', from);
+    const validate = async () => {
+      await this.validateCreator(from);
+
+      let message;
+
+      const isSpokesSet = await this.isSpokesSet();
+      if (!isSpokesSet) {
+        message = `Spokes must be set. isSpokesSet: ${isSpokesSet}`;
+        throw new SetupError(isSpokesSet, message);
+      }
+
+      const isRoutingSet = await this.isRoutingSet();
+      if (isRoutingSet) {
+        message = `Routing is already set. isRoutingSet: ${isRoutingSet}`;
```

Comment:
`const message`

Reply:
Also here, don't add the parameter to the message. In this case (isSpokesSet, etc.) we don't need to pass the boolean arond at all because simply due to the error class we know what's going on. The error class should be `RoutingAlreadySetError` and `SpokesNotSetError`. 

We only want to pass the values if it makes sense to know these values contextually for the implementation and it's not obvious from the type of the error already. So e.g. in case of OutOfBalanceError it makes sense to pass them around, but not in this case. Always ask yourself "Is this value potentially relevant for the implementation and do I have to pass it for th eimplementation to know it?"